cachi2/core/package_managers/yarn/main.py

```diff
@@ -228,19 +232,14 @@ def _generate_environment_variables() -> list[EnvironmentVariable]:
 
 def _verify_corepack_yarn_version(expected_version: semver.Version, source_dir: RootedPath) -> None:
     """Verify that corepack installed the correct version of yarn by checking `yarn --version`."""
-    installed_yarn_version = run_yarn_cmd(
-        ["--version"], source_dir, env={"COREPACK_ENABLE_DOWNLOAD_PROMPT": "0"}
-    ).strip()
-    try:
-        if installed_yarn_version != expected_version:
-            raise PackageManagerError(
-                f"Cachi2 expected corepack to install yarn@{expected_version} but instead "
-                f"found yarn@{installed_yarn_version}."
-            )
-    except ValueError as exc:
```

Comment:
This code extraction should be a separate commit.

Reply:
Done.

Reply:
I'm not sure about the extraction as proposed, i.e.:

1. commit 1 adds a duplicate handler (with a single line changed)
2. commit 2 removes the original code no longer used

instead of
1. move the code to a new standalone helper
2. rest of the changes including class definition

otherwise it's not proposed as an extraction IMO (as a logically isolated change).

Reply:
Good point, I'll restructure it.

Reply:
I may not have been very clear on this especially using the whole hunk as context. Admittedly, there was a slight oversight on my side regarding the version mismatch check which stays in this function, so per our private conversation with @a-ovchinnikov I'll just record the gist of what I meant by splicing that bit of logic:
```
# main.py
-    installed_yarn_version = run_yarn_cmd(
-        ["--version"], source_dir, env={"COREPACK_ENABLE_DOWNLOAD_PROMPT": "0"}
+    installed_yarn_version = extract_yarn_version_from_env(source_dir)
...

# utils.py
+def extract_yarn_version_from_env(source_dir: RootedPath) -> Version:
+    """Extract yarn version from environment."""
+    yarn_version_output = run_yarn_cmd(["--version"], source_dir).strip()
+    return yarn_version_output
```
and then the rest of the changes essentially.

Reply:
Oh, now I see it, thank you! Done.